api/envoy/extensions/filters/http/jwt_authn/v3/config.proto

```diff
@@ -452,6 +453,13 @@ message RemoteJwks {
   //
   //
   config.core.v3.RetryPolicy retry_policy = 4;
+
+  // Refetch JWKS if extracted JWT has no KID or a KID that does not match cached JWKS's KID.
+  // Async JWKS fetching is independent to this refetching process. New JWKS is fetched on
+  // demand and the cache is isolated to the fetched worker thread.
```

Comment:
Saying that the cache is isolated to the worker thread seems like an implementation-specific decision, not really part of the API contract that we should be documenting here.  We are talking about possibly implementing this filter in gRPC in the not-too-distant future, and gRPC would definitely not have per-thread caches.

I suggest either removing this or making it a note to indicate that this is Envoy-specific.

Reply:
That makes sense. FYI https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/jwt_authn/v3/config.proto.html#extensions-filters-http-jwt-authn-v3-remotejwks Under the async_fetch config option, there is existing documentation on how Envoy behaves with respect to fetching JWKS. I extended the docs for this config based on the language provided there to make it clear to the end user. I can update it to make it say Envoy-specific; Just didn't want to confuse users because threads are mentioned in a few places in the JWT filter config docs. Please let me know.

Reply:
Yeah, unfortunately, there's a lot of existing API documentation that references Envoy specifically, since xDS started as part of Envoy.  But since it's really more of a multi-data-plane API these days, we should take the opportunity to generalize the documentation whenever we can.

I think the documentation for any given field should primarily document the API contract in an implementation-agnostic way.  But after that, it's fine to have implementation-specific notes (e.g., "In Envoy, the fetch is done in the main thread" or whatever).